src/Common/src/Internal/NativeFormat/NativeFormatWriter.cs

```diff
@@ -967,6 +973,53 @@ public override bool Equals(object obj)
 #else
     internal
 #endif
+    class FieldSignature : Vertex
+    {
+        private Vertex _containingType;
+        private string _name;
+
+        public FieldSignature(Vertex containingType, string name)
+        {
+            _containingType = containingType;
+            _name = name;
+        }
+
+        internal override void Save(NativeWriter writer)
+        {
+            _containingType.Save(writer);
+            writer.WriteString(_name);
+        }
+
+        public override int GetHashCode()
+        {
+            int hash = 113 + 97 * _containingType.GetHashCode();
+            foreach (char c in _name)
+                hash += (hash << 5) + c * 19;
+
+            return hash;
+        }
+
+        public override bool Equals(object obj)
+        {
+            var other = obj as FieldSignature;
+            if (other == null)
+                return false;
+
+            if (!Object.Equals(other._containingType, _containingType))
+                return false;
+
+            if (!Object.Equals(other._name, _name))
```

Comment:
That's `Equals`, not `ReferenceEquals`. Should be the same as `==` but is consistent with how we do the comparison above...